s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -259,6 +260,102 @@ import akka.util.ByteString
       case _ => downloadRequest
     }
 
+  def makeBucket(bucket: String): Source[Done, NotUsed] =
+    bucketManagementRequest[Done](
+      bucket = bucket,
+      method = HttpMethods.PUT,
+      process = processBucketLifecycleResponse
+    )
+
+  def makeBucketF(bucket: String)(implicit mat: Materializer, attr: Attributes): Future[Done] =
+    makeBucket(bucket).withAttributes(attr).runWith(Sink.ignore)
+
+  def deleteBucket(bucket: String): Source[Done, NotUsed] =
+    bucketManagementRequest[Done](
+      bucket = bucket,
+      method = HttpMethods.DELETE,
+      process = processBucketLifecycleResponse
+    )
+
+  def deleteBucketF(bucket: String)(implicit mat: Materializer, attr: Attributes): Future[Done] =
+    deleteBucket(bucket).withAttributes(attr).runWith(Sink.ignore)
+
+  def checkIfBucketExits(bucketName: String): Source[BucketAccess, NotUsed] =
+    bucketManagementRequest[BucketAccess](
+      bucket = bucketName,
+      method = HttpMethods.HEAD,
+      process = processCheckIfExistsResponse
+    )
+
+  def checkIfBucketExitsF(bucket: String)(implicit mat: Materializer, attr: Attributes): Future[BucketAccess] =
+    checkIfBucketExits(bucket).withAttributes(attr).runWith(Sink.head)
+
+  private def bucketManagementRequest[T](
+      bucket: String,
+      method: HttpMethod,
+      process: (HttpResponse, Materializer) => Future[T]
+  ): Source[T, NotUsed] =
+    Setup
+      .source { mat => implicit attr =>
+        implicit val sys: ActorSystem = mat.system
+        implicit val conf: S3Settings = resolveSettings()
+
+        val location = S3Location(bucket = bucket, key = "")
+
+        signAndRequest(
+          requestHeaders(
+            HttpRequests.bucketManagementRequest(location, method),
+            None
+          )
+        ).mapAsync(1) { response =>
+          process(response, mat)
+        }
+      }
+      .mapMaterializedValue(_ => NotUsed)
+
+  private def processBucketLifecycleResponse(response: HttpResponse, materializer: Materializer): Future[Done] = {
+    import materializer.executionContext
+
+    implicit val mat: Materializer = materializer
+
+    response match {
+      case HttpResponse(status, _, entity, _) if status.isSuccess() =>
+        entity.discardBytes().future()
+      case HttpResponse(_, _, entity, _) =>
+        Unmarshal(entity).to[String].map { err =>
+          throw new S3Exception(err)
+        }
+    }
+  }
+
+  private def processCheckIfExistsResponse(response: HttpResponse, materializer: Materializer): Future[BucketAccess] = {
+    import materializer.executionContext
+
+    implicit val mat: Materializer = materializer
+
+    response match {
+      case HttpResponse(StatusCodes.NotFound, _, entity, _) =>
```

Comment:
This could be just one case for all expected statuses and map to the appropriate result in `map`.